dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -148,6 +183,44 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     }
     CATCH_RETURN()
 
+    HRESULT __stdcall PushNotificationChannel::OnRawNotificationReceived(unsigned int payloadLength, _In_ byte* payload, _In_ HSTRING /*correlationVector */) noexcept
+    {
+        try
+        {
+            BOOL foregroundHandled = false;
+            THROW_IF_FAILED(InvokeAll(payloadLength, payload, &foregroundHandled));
+            THROW_HR(E_INVALIDARG);
+        }
+        catch(...)
+        {
+            //TODO: Capture the HRESULT in telemetry
+            winrt::com_array<uint8_t> payloadArray{ payload, payload + (payloadLength * sizeof(uint8_t)) };
+            std::string commandLine = "----WindowsAppRuntimePushServer:-Payload:\"";
+            commandLine.append(reinterpret_cast<char*>(payload), payloadLength);
+            commandLine.append("\"");
+
+            wil::unique_cotaskmem_string processName;
+            THROW_IF_FAILED(GetCurrentProcessPath(processName));
+
+            const std::string processNameAsUtf8String = ConvertProcessNameToUtf8String(processName.get());
+
+            SHELLEXECUTEINFOA shellExecuteInfo{};
+            shellExecuteInfo.cbSize = sizeof(SHELLEXECUTEINFOA);
+            shellExecuteInfo.fMask = SEE_MASK_NOCLOSEPROCESS | SEE_MASK_DOENVSUBST;
+            shellExecuteInfo.lpFile = processNameAsUtf8String.c_str();
+            shellExecuteInfo.lpParameters = commandLine.c_str();
+
+            shellExecuteInfo.nShow = SW_NORMAL;
```

Comment:
Did you test the case where the foregroundHandled = false? We should ensure that a new background process is launched with the payload.